### PR TITLE
Run some charm tests on Push/PR using Github Actions

### PR DESCRIPTION
**Pull request by jnsgruk**
This PR makes some changes to the CI setup for this repo:

- Renames the existing YAML config for Github Actions from `ci.yaml` to `framework-tests.yaml`
- Adds a suite of tests that clone some of the observability charms, updates their `requirements.txt` files to use the latest commit in a PR/branch, then runs the tests using `tox`
- Adds a suite of tests that runs the test suites from "Hello, Kubecon" and "Hello, Juju!"

The idea here is to catch any low hanging fruit/errors in changes to the framework or `Harness`.

This does imply a dependency on the upstream charm tests actually being correct, but that doesn't stop judgement being applied by the people merging PRs in this repo as to why the tests are failing. Once the Observability have their releasing more defined with CI, we can point at the branch that is associated with `latest/stable` for each charm, where there is a stronger expectation that all tests will pass. The `Hello, *` charms are much more static, and the tests are unlikely to change often.

### CODE REVIEW

**Review by rbarry82 — APPROVED** [review]

File: .github/workflows/framework-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on: [push, pull_request]
+
+jobs:
+  unit-tests:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.5, 3.6, 3.7, 3.8, 3.9]
```
Comment (Vultaire) [inline .github/workflows/framework-tests.yaml:12]: Note: if #607 is merged, 3.5 probably needs to be dropped here

**Review by Vultaire — COMMENTED** [review]
Quick comment re: potential conflict with #607.

**Comment by jnsgruk** [comment]
Agreed, though I think this will merge prior to that PR. We'll just need to be careful to ensure we catch that! Thanks 😊

**Review by niemeyer — APPROVED** [review]
As discussed live, looks good, thanks.

**Pull request merged**